pkg/api/nexus.go

```diff
@@ -30,3 +32,53 @@ func NewRaftReplicator(store db.Store, opts ...raft.Option) (RaftReplicator, err
 		return internal_raft.NewReplicator(store, options), nil
 	}
 }
+
+const RecordSeparator = byte(0x1E)
+
+func (req *SaveRequest) Encode() ([]byte, error) {
+	var argBuf bytes.Buffer
+	if err := gob.NewEncoder(&argBuf).Encode(req.Args); err != nil {
+		return nil, err
+	}
```

Comment:
Shouldn't we use msgpack/protobuf instead?  gob is known to be very slow compared to other encoder/decoders.

Reply:
Yes I considered those alternatives. The issue is that we have a `map[string][]byte` type that needs to be converted to `[]byte` so that the resulting `[]byte` can have nice inter-op with RAFT. Note that we do not need to include `Data` in this encoding process since its already in []byte. Using `protobuf` is just too much of an overhead (proto file, code generation, etc.) just to convert a map to a `[]byte`. Same issue with `msgpack`. Are you aware of anything else ? I was checking [here](https://github.com/alecthomas/go_serialization_benchmarks) but couldn't find anything suitable. Any suggestions ?

Reply:
Did you try https://github.com/vmihailenco/msgpack or https://github.com/shamaton/msgpack. These ones doesn't require code generation.

Reply:
This could be done way simply using proto Marshal/Unmarshal. Have a look at my proposed changes  PR https://github.com/flipkart-incubator/nexus/pull/15

Reply:
Makes sense. Will do this instead.